test/server.py

```diff
@@ -27,6 +28,12 @@
 from hyper.tls import NPN_PROTOCOL
 
 
+class SocketSecure(Enum):
+    SECURE = True
+    INSECURE = False
+    SECURE_NO_AUTO_WRAP = 'NO_AUTO_WRAP'
```

Comment:
Hrm. I suppose? It feels pretty weird though, let's put a comment in here to say that's why we're doing it.